tests/publiccloud/cloud_netconfig.pm

```diff
@@ -12,77 +12,119 @@
+        chomp($local_eth1_ip);
+        my $metadata_eth1_ip = $provider->query_metadata($instance, ifNum => '1', addrCount => '0');
+        die("Failed to get interface 1 IP from metadata server") unless length($metadata_eth1_ip);
+        assert_equals($metadata_eth1_ip, $local_eth1_ip, 'Locally assigned eth1 IP does not equal the IP retrieved from CSP metadata service.');
+
+        record_info('secondary', 'Make sure each interface has also secondary IP address');
+        my $local_eth0_secondary_ip = $instance->ssh_script_output(qq(ip -4 -o a s dev eth0 secondary | grep -Po "inet \\K[\\d.]+" | head -n1));
+        chomp($local_eth0_secondary_ip);
+        my $local_eth1_secondary_ip = $instance->ssh_script_output(qq(ip -4 -o a s dev eth1 secondary | grep -Po "inet \\K[\\d.]+" | head -n1));
+        chomp($local_eth1_secondary_ip);
+
+        record_info('Routes', 'Check that there is default route for each interface');
+        die('There is no IP rule') if ($instance->ssh_script_output('ip route show default dev eth0 | wc -l') == 0);
+        die('There is no IP rule') if ($instance->ssh_script_output('ip route show default dev eth1 | wc -l') == 0);
+
+        record_info('Connections', 'Make HTTPS connection from each interface and check they have different public IP address');
+        $instance->ssh_assert_script_run('curl -s https://wtfismyip.com/text');
+        my $eth0_public_ip = $instance->ssh_script_output('curl --fail --interface eth0 -s https://wtfismyip.com/text');
+        my $eth1_public_ip = $instance->ssh_script_output('curl --fail --interface eth1 -s https://wtfismyip.com/text');
+        assert_not_equals($eth0_public_ip, $eth1_public_ip, "The connection from eth0 shouldn't have the same public IPv4 address as connection from eth1");
+
+        record_info('Rules', 'Make sure there are IP rules for each IP address');
+        die('There is no IP rule') if ($instance->ssh_script_output("ip rule list all from $local_eth0_ip | wc -l") == 0);
+        die('There is no IP rule') if ($instance->ssh_script_output("ip rule list all from $local_eth0_secondary_ip | wc -l") == 0);
+        die('There is no IP rule') if ($instance->ssh_script_output("ip rule list all from $local_eth1_ip | wc -l") == 0);
+        die('There is no IP rule') if ($instance->ssh_script_output("ip rule list all from $local_eth1_secondary_ip | wc -l") == 0);
+
+        record_info('Neighbors', 'Make sure there is at least one ARP neighbor on each interface');
+        die('There are no ARP neighbors on eth0') if ($instance->ssh_script_output("ip neighbor show dev eth0 | wc -l") == 0);
+        die('There are no ARP neighbors on eth1') if ($instance->ssh_script_output("ip neighbor show dev eth1 | wc -l") == 0);
+
+        record_info('Local removal', 'Remove eth0 secondary address and eth1 default route and check if it reappears.');
```

Comment:
I don't see the point of this check for cloud_netconfig. This is testing something different IMHO and could be removed. What do you think? :thinking:

Reply:
From https://github.com/SUSE-Enceladus/cloud-netconfig/blob/master/README.md:
> Interface configurations will be checked periodically on each DHCP lease renewal and additionally, if the systemd timer is enabled, every 60 seconds. cloud-netconfig detects changes in the metadata configuration and updates interface configurations and routing policies accordingly. This means that IP addresses and ranges that were removed from the virtual interface configuration will be removed from the interface, but only addresses and ranges that were automatically added by cloud-netconfig will be removed. Addresses added manually by the administrator or by another tool (e.g. high-availability software) will not be touched.

From my understanding `cloud-netconfig` periodically checks if the interfaces and addresses defined in cloud metadata are present in the system. This checks that by removing some and checking if they got reapplied.

Reply:
I see, yes then it makes sense :+1: